enterprise/users/src/main/java/io/crate/protocols/http/HttpAuthUpstreamHandler.java

```diff
@@ -140,24 +145,30 @@ boolean authorized() {
     }
 
     @VisibleForTesting
-    static String userFromRequest(HttpRequest request, @Nullable SSLSession session, Settings settings) {
-        if (request.headers().contains(AuthSettings.HTTP_HEADER_USER)) {
-            return request.headers().get(AuthSettings.HTTP_HEADER_USER);
+    static Tuple<String, SecureString> credentialsFromRequest(HttpRequest request, @Nullable SSLSession session, Settings settings) {
+        String username = null;
+        if (request.headers().contains(AuthSettings.HTTP_HEADER_BASIC_AUTH)) {
+            // Prefer Http Basic Auth
+            return RestSQLAction.extractCredentialsFromHttpBasicAuthHeader(
+                request.headers().get(AuthSettings.HTTP_HEADER_BASIC_AUTH));
+        } else if (request.headers().contains(AuthSettings.HTTP_HEADER_USER)) {
+            // Fallback to deprecated setting
+            username = request.headers().get(AuthSettings.HTTP_HEADER_USER);
         } else {
             // prefer commonName as userName over AUTH_TRUST_HTTP_DEFAULT_HEADER user
             if (session != null) {
                 try {
                     Certificate certificate = session.getPeerCertificates()[0];
-                    String commonName = SSL.extractCN(certificate);
-                    if (commonName != null) {
-                        return commonName;
```

Comment:
isn't this a behaviour change? E.g. if there is a cert without commonName it would now fallback to the AUTH_TRUST_HTTP_HEADER instead of failing?

Reply:
Nevermind, I only glanced at it and miss-read it.